From a234ec4dfc10f5533698e7d879963ffd9bf424a3 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Wed, 11 Nov 2020 11:57:35 +0200
Subject: [PATCH] [SW-14676] fit: remove negative logic

Checking the validity of FIT image is done in negative logic. This is
not needed since the .bss section is writable
(unlike u-boot which in we must use .data section for writeable globals)

Change-Id: I7d5953c23bf3307f05ad8d04b3c0817ed3144413
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/preboot/module_lkd.c b/preboot/module_lkd.c
index 508af85..855bef7 100644
--- a/preboot/module_lkd.c
+++ b/preboot/module_lkd.c
@@ -14,13 +14,11 @@
 #include "fit.h"
 
 struct lkd_status {
-	int fit_not_checked;	/* negative logic */
+	bool is_fit_checked;
 	u32 msg_reg_val;
 };
 
-static struct lkd_status sts = {
-	.fit_not_checked = 1,
-};
+static struct lkd_status sts;
 
 void module_lkd_setup(void)
 {
@@ -32,7 +30,7 @@
 	enum cmd_check_res res;
 
 	/* we already have the FIT and it is good */
-	if (sts.fit_not_checked != 1)
+	if (sts.is_fit_checked)
 		return FOUND_NO_RETURN;
 
 	sts.msg_reg_val = cpu_boot_cmd_get();
@@ -68,7 +66,7 @@
 
 void module_lkd_run(void)
 {
-	sts.fit_not_checked = !!fit_load_images();
+	sts.is_fit_checked = !fit_load_images();
 }
 
 void module_lkd_run_noret(void)
